### PR TITLE
MM-50073 : Infinite recursion when opening New Category modal from Channel Move To menu

### PR DESCRIPTION
**Pull request by M-ZubairAhmed**
#### Summary
- Added manual focustrap functionality in generic modals
- Added manual focustrap in edit category modal

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-50073

#### Related Pull Requests
Child of https://github.com/mattermost/mattermost-webapp/pull/12127

#### Screenshots
n/a

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: components/edit_category_modal/edit_category_modal.tsx [hunk 1]
```diff
@@ -4,14 +4,17 @@
 import React from 'react';
 import {FormattedMessage} from 'react-intl';
 
+import {GenericModal} from '@mattermost/components';
```
Comment (hmhealey) [inline components/edit_category_modal/edit_category_modal.tsx:7]: Is this change needed or can we still use the old one? It seems like the one from `components/generic_modal` includes some CSS that we might otherwise miss when importing from the components package right now

File: packages/components/src/generic_modal/generic_modal.tsx [hunk 2]
```diff
@@ -31,10 +33,17 @@ export type Props = {
     backdrop?: boolean;
     backdropClassName?: string;
     children: React.ReactNode;
+
+    /**
+     * This is temporary fix until we update underlying package of generic modal.
+     * @caution Use this with enforceFocus={false}
+     */
+    shouldFocusTrapHandleFocus?: boolean;
```
Comment (hmhealey) [inline packages/components/src/generic_modal/generic_modal.tsx:41]: Instead of making it so that people using this also need to set `enforceFocus={false}`, could we instead make  it so that this component automatically passes `enforceFocus={false}` into the React Boostrap `Modal` automatically when this is set to true?

Comment (hmhealey) [reply packages/components/src/generic_modal/generic_modal.tsx:41]: Alternatively, why not always have this enabled whenever `enforceFocus` is true (or unset since it defaults to true)? Are there any cases that we wouldn't want this behaviour?

**Review by hmhealey — CHANGES_REQUESTED** [review]
It looks like you forgot to include the files for `FocusTrap` as well

File: components/edit_category_modal/edit_category_modal.tsx (hunk 1 above)
Comment (M-ZubairAhmed) [reply components/edit_category_modal/edit_category_modal.tsx:7]: Yeah reverted that

File: packages/components/src/generic_modal/generic_modal.tsx (hunk 2 above)
Comment (M-ZubairAhmed) [reply packages/components/src/generic_modal/generic_modal.tsx]: thats a good idea, made the change

File: components/edit_category_modal/edit_category_modal.tsx (hunk 1 above)
Comment (M-ZubairAhmed) [reply components/edit_category_modal/edit_category_modal.tsx]: reverted to old one

File: packages/components/src/generic_modal/generic_modal.tsx (hunk 2 above)
Comment (M-ZubairAhmed) [reply packages/components/src/generic_modal/generic_modal.tsx]: yes thats a seems like a good idea, i have used existing prop `enforceFocus` to activate the focus trap

**Review by pvev — APPROVED** [review]
LGTM! thanks @M-ZubairAhmed 👍

**Review by jgilliam17 — APPROVED** [review]
Thanks @M-ZubairAhmed 
Tested, looks good to merge
Verified opening modals from sidebar menus works as expected. 
[E2E report ](https://mattermost-cypress-report.s3.amazonaws.com/298969-92b386a-onprem-ent-webapp-pr-12128-mm-ee-test:92b386a/mochawesome.html)- no PR related failures

**Review by hmhealey — APPROVED** [review]
